### ISSUE TITLE
Getting some errors in logs

### ISSUE THREAD

**Issue by akolov** [nitaybz/homebridge-sensibo-ac#61]
Hey, first of all would like to say huge thanks for developing and maintaining this plugin. Can't imagine how I'd get by without it.

Also, I've started getting quite some errors in logs and status of ACs in the Home app won't update for a while. I'm using v2.1.8.
```
[6/20/2021, 1:41:58 AM] [SensiboAC] Adding HeaterCooler Service in the Living Room
(node:29908) UnhandledPromiseRejectionWarning: AssertionError [ERR_ASSERTION]: characteristic prop validValues cannot be empty array
    at TargetHeaterCoolerState.Characteristic.setProps (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1120:15)
    at AirConditioner.addHeaterCoolerService (/usr/local/lib/node_modules/homebridge-sensibo-ac/homekit/AirConditioner.js:129:5)
    at new AirConditioner (/usr/local/lib/node_modules/homebridge-sensibo-ac/homekit/AirConditioner.js:77:8)
    at /usr/local/lib/node_modules/homebridge-sensibo-ac/sensibo/syncHomeKitCache.js:18:28
    at Array.forEach (<anonymous>)
    at SensiboACPlatform.syncHomeKitCache (/usr/local/lib/node_modules/homebridge-sensibo-ac/sensibo/syncHomeKitCache.js:10:20)
    at Timeout._onTimeout (/usr/local/lib/node_modules/homebridge-sensibo-ac/sensibo/refreshState.js:68:14)
(node:29908) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
```

```
[6/20/2021, 1:44:53 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:44:53 AM] [homebridge-sensibo-ac] Error: This callback function has already been called by someone else; it can only be called one time.
    at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/once.ts:6:13
    at CurrentHeaterCoolerState (/usr/local/lib/node_modules/homebridge-sensibo-ac/homekit/StateManager.js:66:6)
    at CurrentHeaterCoolerState.emit (events.js:327:22)
    at /usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1508:14
    at new Promise (<anonymous>)
    at CurrentHeaterCoolerState.<anonymous> (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:1506:12)
    at step (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:143:27)
    at Object.next (/usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:124:57)
    at /usr/local/lib/node_modules/homebridge/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
```

**Comment by akolov** [comment]
Some more logs:
```
[6/20/2021, 4:30:03 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:30:03 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:30:03 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:30:18 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:30:18 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:30:18 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 5:39:09 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 5:39:09 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 5:39:09 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 5:42:31 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 5:42:31 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 5:42:31 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 6:59:34 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 6:59:34 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 6:59:34 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 7:00:32 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 7:00:32 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 7:00:32 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 9:11:02 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 9:11:02 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 9:11:02 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 11:20:37 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 11:20:37 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 11:20:37 AM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 12:33:16 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 12:33:16 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 12:33:16 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 12:58:46 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 12:58:46 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 12:58:46 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:15:25 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:15:25 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:15:25 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:18:25 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:18:25 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 1:18:25 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 2:26:32 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 2:26:32 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 2:26:32 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:18:49 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:18:49 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 4:18:49 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:03:56 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:03:56 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:03:56 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:03:56 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:03:56 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:04:42 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:04:42 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:06:45 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:06:45 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:06:45 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:06:53 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:06:53 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:49:15 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'S2R2': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:49:15 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Active': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
[6/20/2021, 8:49:15 PM] [homebridge-sensibo-ac] This plugin threw an error from the characteristic 'Current Heater-Cooler State': Unhandled error thrown inside read handler for characteristic: This callback function has already been called by someone else; it can only be called one time.. See https://git.io/JtMGR for more info.
```

**Comment by nitaybz** [comment]
I guess it all appears because the lack of support in Sensibo Pure.... am I right? did you add Pure recently?

**Comment by akolov** [comment]
Ah you're right, I did. Didn't expect it to explode, since there's no mention of pure anywhere.

**Comment by nitaybz** [comment]
I'm on it... will be fixed pretty soon

**Comment by nitaybz** [comment]
you can already try it, install beta version from homebridge UI by clicking on the plugin's "install previous versions" button... you'll find the recent beta there... it should solve the issue.
please report back

**Comment by akolov** [comment]
It works now. Throwing warnings, attached below, but no problems controlling ACs via HomeKit now.
Thanks a lot!

```
[6/23/2021, 9:50:57 AM] [homebridge-sensibo-ac] This plugin generated a warning from the characteristic 'On': characteristic was supplied illegal value: null! Home App will reject null for Apple defined characteristics. See https://git.io/JtMGR for more info.
[6/23/2021, 9:51:04 AM] [homebridge-sensibo-ac] This plugin generated a warning from the characteristic 'On': characteristic was supplied illegal value: null! Home App will reject null for Apple defined characteristics. See https://git.io/JtMGR for more info.
[6/23/2021, 9:51:11 AM] [homebridge-sensibo-ac] This plugin generated a warning from the characteristic 'On': characteristic was supplied illegal value: null! Home App will reject null for Apple defined characteristics. See https://git.io/JtMGR for more info.
[6/23/2021, 9:51:27 AM] [homebridge-sensibo-ac] This plugin generated a warning from the characteristic 'On': characteristic was supplied illegal value: null! Home App will reject null for Apple defined characteristics. See https://git.io/JtMGR for more info.
```

**Comment by nitaybz** [comment]
I think I fixed everything in version 2.2

Please verify

**Comment by akolov** [comment]
Works perfectly and better support of Pure than official HomeKit integration. Kudos!
